src/common.rs

```diff
@@ -355,6 +355,130 @@ pub struct LoadAvg {
     pub fifteen: f64,
 }
 
+#[cfg(not(target_os = "windows"))]
+pub mod uid {
+    use std::ops::Deref;
+
+    /// A user id wrapping a platform specific type
+    ///
+    /// ```no_run
+    /// use sysinfo::{System, SystemExt, UserExt};
+    ///
+    /// let mut s = System::new_all();
+    /// for user in s.get_users() {
+    ///     println!("{}", *user.get_uid());
+    /// }
+    #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+    pub struct Uid(libc::uid_t);
+
+    impl Uid {
+        pub(crate) fn new(val: libc::uid_t) -> Self {
+            Self(val)
+        }
+    }
+
+    impl Deref for Uid {
+        type Target = libc::uid_t;
+
+        fn deref(&self) -> &Self::Target {
+            &self.0
+        }
+    }
+}
+
+#[cfg(target_os = "windows")]
+pub mod uid {
+    use std::ops::Deref;
+
+    /// A user id wrapping a platform specific type
+    ///
+    /// ```no_run
+    /// use sysinfo::{System, SystemExt, UserExt};
+    ///
+    /// let mut s = System::new_all();
+    /// for user in s.get_users() {
+    ///     println!("{}", *user.get_uid());
+    /// }
+    #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+    pub struct Uid(u32);
+
+    impl Uid {
+        pub(crate) fn new(val: u32) -> Self {
```

Comment:
There isn't much point for this method actually. You can just write `Uid(whatever)` and it's even shorter than calling the constructor haha.

Reply:
Removed.